core/src/payload.ts

```diff
@@ -57,12 +61,12 @@ export function isNonEmptyJson(property): boolean {
 /**
  * Is property a JSON?
  */
-export function isJson(property: Object): boolean {
-	return (typeof property !== 'undefined' && property !== null &&
-	(property.constructor === {}.constructor || property.constructor === [].constructor));
+export function isJson(property: unknown): boolean {
+	const record = property as Record<string, unknown>;
```

Comment:
Should this be `property as Record<string, unknown> | undefined | null`?  Based on the checks you still on line 66.

And is there a reason to reassign `record = property` instead of changing the signature of the function to accept a `Record<string, unknown>`?

Both genuine questions, not complaints!

Reply:
I can't change the signature to `Record<string, unknown>` as this function should be allowed to be passed anything and checks if it is valid json (and there is a weird test that passes it a function). 
As the tests are in TypeScript, making it `Record<string, unknown>` means the test doesn't compile so I think to be true to its real usecase then it should stay as `unknown`. 
It's a bit of an odd function for a typescript library in reality, pretty much exists to sanitise nonsense from any JavaScript consumers.
As for `property as Record<string, unknown> | undefined | null`, yeah I think that's a more explicit approach for sure 👍